pytorch_lightning/accelerators/accelerator.py

```diff
@@ -44,15 +45,29 @@ class Accelerator:
     One to handle differences from the training routine and one to handle different precisions.
     """
 
-    def __init__(self, precision_plugin: PrecisionPlugin, training_type_plugin: TrainingTypePlugin) -> None:
+    def __init__(
+        self, training_type_plugin: TrainingTypePlugin, precision_plugin: Optional[PrecisionPlugin] = None
```

Comment:
This change is not backward compatible, as the order is reversed now compared to before.

The only solution would be to set a token default value for training type plugin.

On a tangential note, I don't get the point for deprecating just the constructor, either the API is stable or it isn't, doing it for the constructor seems arbitrary.

Reply:
Reviewers need to agree here or we can't move forward. 

Either we add deprecation and backward compatibility for a few commits for all the 0.0001% users who actually install master and use these APIs, OR we remove the reference here, breaking change, and move on.

> either the API is stable or it isn't

The API is not stable and we know that from the docs. We prioritize making it stable for 1.6.

Reply:
Hey @awaelchli. Adding to @carmocca  point, let's add the deprecation warning and not change the init API. Better to do it at once when we actually break the API.

Reply:
Sorry if this wasn't clear, I'm fine with both deprecating OR not deprecating, I was just raising a question. I don't want to block the PR over that.

but IF the deprecation is chosen, then my point about setting a token default value stands.

Reply:
@carmocca Is the current way sufficient?